src/main/kotlin/kweb/util/misc.kt

```diff
@@ -22,16 +25,42 @@ fun createNonce(length: Int = 6): String {
     return Base64.getUrlEncoder().encodeToString(ar).substring(0, length)
 }
 
-val gson = Gson()
-
 val scheduledExecutorService: ScheduledExecutorService = Executors.newScheduledThreadPool(5)
 
 fun String.escapeEcma() = StringEscapeUtils.escapeEcmaScript(this)!!
 
+val gson = Gson()
+//TODO I'm not sure what the best way to deal with this is.
 fun Any.toJson(): String = gson.toJson(this)
 
 data class JsFunction(val jsId: Int, val arguments: List<Any?> = emptyList())
 
+fun primitiveToJson(value: Any?, errorMsg: String = "Argument is required to be String or primitive type"): JsonElement {
+    return when(value) {
```

Comment:
Should we be using: https://kotlin.github.io/kotlinx.serialization/kotlinx-serialization-json/kotlinx-serialization-json/kotlinx.serialization.json/encode-to-json-element.html ?

Reply:
If I try to write this function as 
fun primitiveToJson(value: Any?) {//abbreviated 
return Json.encodeToJsonElement(value)
}
it causes errors saying that it can't find a serializer for Any.

Weirdly, I have
fun Any.toJson(): String = Json.encodeToJsonElement(this).toString()
This compiles fine, and the tests and todo app still work.

I'm not sure why that works in one place and doesn't in another. Maybe because value is a nullable Any, in primitiveToJson().

Reply:
Hmm, perhaps this? https://kotlin.github.io/kotlinx.serialization/kotlinx-serialization-json/kotlinx-serialization-json/kotlinx.serialization.json/encode-to-dynamic.html